src/lib/Libifl/pbsD_submit.c

```diff
@@ -196,19 +195,23 @@ __pbs_submit(int c, struct attropl  *attrib, char *script, char *destination, ch
 	}
 
 	/* initiate the queueing of the job */
-
 	for (pal = attrib; pal; pal = pal->next)
 		pal->op = SET;		/* force operator to SET */
 
 	cred_info = (struct cred_info *) ptr->th_cred_info;
 
 	if ((!script || (*script == '\0')) && (!cred_info || (cred_info->cred_len <= 0))) {
 		/* no cred and no script, let's request implicit commit to cut one message exchange */
-		if (!(extend = pbs_strcat(&extend, NULL, EXTEND_OPT_IMPLICIT_COMMIT))) {
-			if (set_conn_errtxt(c, "Failed to allocated memory") != 0)
-				pbs_errno = PBSE_SYSTEM;
-			goto error;
-		}
+		if (extend) {
+			lextend = EXTEND_OPT_IMPLICIT_COMMIT;
+			if (!pbs_strcat(&lextend, NULL, extend)) {
+				if (set_conn_errtxt(c, "Failed to allocate memory") != 0)
+					pbs_errno = PBSE_SYSTEM;
+				goto error;
```

Comment:
I think it will crash in this case because lextend is a string literal.

Reply:
Hi @arungrover i think it will not crash and i did test it under valgrind and without valgrind as well. I think what happens is that lextend points to a literal to start with. However, inside the pbs_strcat() we call realloc(lextend) right? Realloc merely allocated a new memory location and does a memmove internally to copy over the stuff to the new location. This does not change the address of the literal at all.